src/parameterizations/lateral/MOM_lateral_mixing_coeffs.F90

```diff
@@ -470,6 +535,74 @@ subroutine calc_resoln_function(h, tv, G, GV, US, CS)
 
 end subroutine calc_resoln_function
 
+!> Calculates and stores functions of SQG mode
+subroutine calc_sqg_struct(h, tv, G, GV, US, CS, dt, MEKE)
+  type(ocean_grid_type),                     intent(inout) :: G  !< Ocean grid structure
+  type(verticalGrid_type),                   intent(in)    :: GV !< Vertical grid structure
+  type(unit_scale_type),                     intent(in)    :: US !< A dimensional unit scaling type
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), intent(in)    :: h  !< Layer thickness [H ~> m or kg m-2]
+  type(thermo_var_ptrs),                    intent(in)     :: tv !<Thermodynamic variables
+   real,                                      intent(in)    :: dt !< Time increment [T ~> s]
+  type(VarMix_CS),                           intent(inout) :: CS !< Variable mixing control struct
+  type(MEKE_type),                           intent(in)     :: MEKE !< MEKE struct
+
+  ! Local variables
+  real, dimension(SZI_(G), SZJ_(G),SZK_(GV)+1) :: &
+    e             ! The interface heights relative to mean sea level [Z ~> m].
+  real, dimension(SZIB_(G), SZJ_(G),SZK_(GV)+1) :: N2_u ! Square of Brunt-Vaisala freq at u-points [L2 Z-2 T-2 ~> s-2]
+  real, dimension(SZI_(G), SZJB_(G),SZK_(GV)+1) :: N2_v ! Square of Brunt-Vaisala freq at v-points [L2 Z-2 T-2 ~> s-2]
+  real, dimension(SZIB_(G), SZJ_(G),SZK_(GV)+1) :: dzu ! Z-thickness at u-points [Z ~> m]
+  real, dimension(SZI_(G), SZJB_(G),SZK_(GV)+1) :: dzv ! Z-thickness at v-points [Z ~> m]
+  real, dimension(SZIB_(G), SZJ_(G),SZK_(GV)+1) :: dzSxN ! |Sx| N times dz at u-points [Z T-1 ~> m s-1]
+  real, dimension(SZI_(G), SZJB_(G),SZK_(GV)+1) :: dzSyN ! |Sy| N times dz at v-points [Z T-1 ~> m s-1]
+  real, dimension(SZI_(G), SZJ_(G)) :: f  ! Absolute value of the Coriolis parameter at h point [T-1 ~> s-1]
+  real :: N2  ! Positive buoyancy frequency square or zero [L2 Z-2 T-2 ~> s-2]
+  real :: dzc  ! Spacing between two adjacent layers in stretched vertical coordinate [m]
+  real, dimension(SZI_(G), SZJ_(G)) :: Le  ! Eddy length scale [m]
+  integer :: i, j, k, is, ie, js, je, nz
+
+  is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
+
+  if (.not. CS%initialized) call MOM_error(FATAL, "MOM_lateral_mixing_coeffs.F90, calc_slope_functions: "//&
+         "Module must be initialized before it is used.")
+
+  call find_eta(h, tv, G, GV, US, e, halo_size=2)
+  call calc_isoneutral_slopes(G, GV, US, h, e, tv, dt*CS%kappa_smooth, CS%use_stanley_iso, &
+                                  CS%slope_x, CS%slope_y, N2_u=N2_u, N2_v=N2_v,dzu=dzu, dzv=dzv, &
+                                  dzSxN=dzSxN, dzSyN=dzSyN, halo=1)
+
+  do j=js,je ; do i=is,ie
+    CS%sqg_struct(i,j,1) = 1.0
+  enddo ; enddo
+  if (allocated(MEKE%Le)) then
+    do j=js,je ; do i=is,ie
+      Le(i,j) = MEKE%Le(i,j)
+      f(i,j) = max(0.25 * abs((G%CoriolisBu(I,J) + G%CoriolisBu(I-1,J-1)) + &
+                       (G%CoriolisBu(I-1,J) + G%CoriolisBu(I,J-1))), 1.0e-8 * US%s_to_T)
```

Comment:
The hard-coded dimensional value of 1.0e-8 s-1 is not a particularly small number to use as a floor on the rotation.  What is the justification for this value, and could it be replaced by something else, such as a miniscule fraction of the planetary rotation rate omega?